x-pack/plugins/apm/public/components/shared/environment_select/index.tsx

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import moment from 'moment';
+import { i18n } from '@kbn/i18n';
+import React, { useMemo, useState, useCallback } from 'react';
+import { debounce } from 'lodash';
+import { EuiComboBox, EuiComboBoxOptionOption } from '@elastic/eui';
+import { getEnvironmentLabel } from '../../../../common/environment_filter_values';
+import { SERVICE_ENVIRONMENT } from '../../../../common/elasticsearch_fieldnames';
+import { FETCH_STATUS, useFetcher } from '../../../hooks/use_fetcher';
+import { Environment } from '../../../../common/environment_rt';
+
+export function EnvironmentSelect({
+  environment,
+  environmentOptions,
+  status,
+  serviceName,
+  start,
+  end,
+  onChange,
+}: {
+  environment: Environment;
+  environmentOptions: Array<EuiComboBoxOptionOption<string>>;
```

Comment:
Maybe this should be named `availableEnvironments: Environment[]`? the option formatting sounds like it's an implementation detail of `EnvironmentSelect`.